examples/wmt/train.py

```diff
@@ -497,13 +485,53 @@ def tohost(x):
   return np.array(x).reshape((n_device * n_batch,) + tuple(remaining_dims))
 
 
-def main(argv):
-  if len(argv) > 1:
-    raise app.UsageError('Too many command-line arguments.')
-
-  if FLAGS.jax_backend_target:
+def train_and_evaluate(
+  random_seed, jax_backend_target, model_dir, data_dir, vocab_path, vocab_size,
+  dataset_name, eval_dataset_name, batch_size, beam_size, eval_frequency,
+  num_train_steps, num_eval_steps, learning_rate, warmup_steps,
+  label_smoothing, weight_decay, max_target_length, max_eval_target_length,
+  max_predict_length, emb_dim, num_heads, num_layers, qkv_dim, mlp_dim,
+  share_embeddings, logits_via_embedding, use_bfloat16, restore_checkpoints,
+  save_checkpoints, checkpoint_freq):
+  """Executes model training and evaluation loop.
+  
+  Args:
```

Comment:
Could we instead use the newly open-sourced `ml_collections.ConfigDict` ?

Reply:
I think it will be better to decouple the changes. Similar thoughts by Avital on #419

Reply:
Decoupling SGTM. I'm not a big fan of these huge signatures. If we refactored the examples first to use `ConfigDict` and then added the unit tests, we would never have to introduce them. Feel free to decide yourself in which order you prefer to do the two changes...

Reply:
Yeah agreed. My thinking behind doing this before using ConfigDicts was that we will have a unit test in place as a sanity check for the transition from flags to configs. WDYT? 

I was leaning towards this solution unit test -> follow up PR with ConfigDicts.